graviti/portex/field.py

```diff
@@ -11,11 +11,95 @@
 
 from graviti.portex.base import _INDENT, PortexType
 from graviti.portex.package import Imports
-from graviti.utility import NameOrderedDict
+from graviti.utility import InmutableNameOrderedDict, NameOrderedDict
 
 _T = TypeVar("_T", bound="Fields")
 
 
+class InmutableFields(InmutableNameOrderedDict[PortexType]):
+    """Represents an inmutable fields dict."""
+
+    def __repr__(self) -> str:
+        return self._repr1(0)
+
+    @classmethod
+    def _check_value(cls, value: PortexType) -> None:
+        if not isinstance(value, PortexType):
+            raise TypeError(f'The value in "{cls.__name__}" should be a PortexType')
+
+    def _repr1(self, level: int) -> str:
+        indent = level * _INDENT
+        lines = ["{"]
+        for name, portex_type in self.items():
+            lines.append(
+                f"{_INDENT}'{name}': "  # pylint: disable=protected-access
+                f"{portex_type._repr1(level + 1)},"
+            )
+
+        lines.append("}")
+        return f"\n{indent}".join(lines)
+
+    def _setitem(self, key: str, value: PortexType) -> None:
+        self._check_value(value)
+        super()._setitem(key, value)
+
+
+class MutableFields(NameOrderedDict[PortexType], InmutableFields):
+    """Represents a mutable fields dict."""
+
+    def insert(self, index: int, name: str, portex_type: PortexType) -> None:
+        """Insert the name and portex_type at the index.
+
+        Arguments:
+            index: The index to insert the field.
+            name: The name of the field to be inserted.
+            portex_type: The portex_type of the field to be inserted.
+
+        Raises:
+            KeyError: When the name already exists in the Fields.
+
+        """
+        self._check_key(name)
+        self._check_value(portex_type)
+
+        if self.__contains__(name):
+            raise KeyError(f'"{name}" already exists in the Fields')
+
+        self._keys.insert(index, name)
+        self._data.__setitem__(name, portex_type)
+
+    def astype(self, name: str, portex_type: PortexType) -> None:
```

Comment:
what if user use `__setitem__` to change type of fields?

Reply:
It is also allowed.